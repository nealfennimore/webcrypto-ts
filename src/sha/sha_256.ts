import * as alg from "../alg.js";
import { WebCrypto } from "../crypto.js";
import { Sha256ArrayBuffer, ShaShared } from "./shared.js";

export namespace SHA_256 {
    export const digest = async (data: BufferSource) =>
        WebCrypto.digest<Sha256ArrayBuffer>(alg.SHA.Variant.SHA_256, data);

    export const hexify = (digest: Sha256ArrayBuffer) =>
        ShaShared.hexify(digest);
}
