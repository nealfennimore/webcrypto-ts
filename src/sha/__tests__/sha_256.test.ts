import { SHA_256 } from "../sha_256";

describe("SHA_256", () => {
    it("should hash", async () => {
        const data = encode("a message");
        const hash = await SHA_256.digest(data);
        expect(hash.byteLength).toEqual(32);
    });
    it("should hexify", async () => {
        const data = encode("a message");
        const hash = await SHA_256.digest(data);
        expect(SHA_256.hexify(hash)).toMatchSnapshot();
    });
});
