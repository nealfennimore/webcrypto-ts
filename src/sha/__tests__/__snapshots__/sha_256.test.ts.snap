// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SHA_256 should hexify 1`] = `"f53c09ca39717a45c62d9aca8f8113eddbfd5f81dcab0b33b1c1834075225e68"`;
